Assets/Scripts/Wall/WallSegment.cs

```diff
@@ -22,71 +22,74 @@ public class WallSegment : MonoBehaviour
         public bool chosenOne;
 
         private MeshFilter _meshFilter;
+
         private void Start()
         {
             _meshFilter = GetComponentInChildren<MeshFilter>();
             normalWall = _meshFilter.mesh;
             // UpdateSoldierState();
         }
 
-        public void RepairWall()
-        {
-            health = Mathf.Min(2, health + 1);
-            ChangeWallState();
-        }
-
-        public void DamageWall()
-        {
-            health = Mathf.Max(0, health - 1);
-            ChangeWallState();
-        }
-
         private void ChangeWallState()
         {
             _meshFilter.mesh = health switch
             {
                 0 => destroyedWall,
                 1 => damagedWall,
-                2 => normalWall, 
+                2 => normalWall,
                 _ => _meshFilter.mesh
             };
         }
 
+        public void DamageScaffolding()
+        {
+            print("TEST");
+            isScaffoldingIntact = false;
+            scaffoldingPiece.SetActive(false);
+            if (isSoldierPresent)
+            {
+                soldier.Die();
+                isSoldierPresent = false;
+            }
+        }
+
         public void RepairScaffolding()
```

Comment:
Same thing as the wall. RepairScaffolding should return whether or not the repair was successful